StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheetFlowControllerTests.swift

```diff
@@ -0,0 +1,166 @@
+        let displayData = PaymentSheet.FlowController.PaymentOptionDisplayData(
+            paymentOption: paymentOption,
+            currency: "usd",
+            iconStyle: .filled
+        )
+
+        // Test labels for US bank account without linked bank
+        XCTAssertEqual(displayData.labels.label, "US bank account")
+        XCTAssertEqual(displayData.labels.sublabel, "US bank account")
+    }
+
+    func testPaymentOptionDisplayData_SavedCardLabels() {
+        // Create a saved Visa card payment method using test helper
+        let paymentMethod = STPPaymentMethod._testCard()
+
+        let paymentOption = PaymentSheet.PaymentOption.saved(paymentMethod: paymentMethod, confirmParams: nil)
+        let displayData = PaymentSheet.FlowController.PaymentOptionDisplayData(
+            paymentOption: paymentOption,
+            currency: "usd",
+            iconStyle: .filled
+        )
+
+        // Test labels for saved card
+        XCTAssertEqual(displayData.labels.label, "Visa")
+        XCTAssertEqual(displayData.labels.sublabel, "•••• 4242")
+    }
+
+    func testPaymentOptionDisplayData_SavedUSBankAccountLabels() {
+        // Create a saved US bank account payment method using test helper
+        let paymentMethod = STPPaymentMethod._testUSBankAccount()
+
+        let paymentOption = PaymentSheet.PaymentOption.saved(paymentMethod: paymentMethod, confirmParams: nil)
+        let displayData = PaymentSheet.FlowController.PaymentOptionDisplayData(
+            paymentOption: paymentOption,
+            currency: "usd",
+            iconStyle: .filled
+        )
+
+        // Test labels for saved US bank account - should show bank name from saved payment method
+        XCTAssertEqual(displayData.labels.label, "STRIPE TEST BANK")
+        XCTAssertEqual(displayData.labels.sublabel, "••••6789")
+    }
+
+    func testPaymentOptionDisplayData_ApplePayLabels() {
+        let paymentOption = PaymentSheet.PaymentOption.applePay
+        let displayData = PaymentSheet.FlowController.PaymentOptionDisplayData(
+            paymentOption: paymentOption,
+            currency: "usd",
+            iconStyle: .filled
+        )
+
+        // Test labels for Apple Pay
+        XCTAssertEqual(displayData.labels.label, "Apple Pay")
+        XCTAssertEqual(displayData.labels.sublabel, nil)
+    }
+
+    func testPaymentOptionDisplayData_LinkLabels() {
+        let linkOption = PaymentSheet.LinkConfirmOption.wallet
+        let paymentOption = PaymentSheet.PaymentOption.link(option: linkOption)
+        let displayData = PaymentSheet.FlowController.PaymentOptionDisplayData(
+            paymentOption: paymentOption,
+            currency: "usd",
+            iconStyle: .filled
+        )
+
+        // Test labels for Link
+        XCTAssertEqual(displayData.labels.label, STPPaymentMethodType.link.displayName)
+        XCTAssertEqual(displayData.labels.sublabel, STPPaymentMethodType.link.displayName)
```

Comment:
We should only show `Link` as the label in this case, not for both label and sublabel.